unity-renderer/Assets/Scripts/MainScripts/DCL/Controllers/ExploreV2/Scripts/Sections/PlacesAndEventsSection/SubSections/HighlightsSubSection/HighlightsSubSectionMenu/HighlightsSubSectionComponentController.cs

```diff
@@ -130,63 +134,66 @@ internal void RequestAllPlacesAndEvents()
-            {
-                placesFromAPI = placeList;
-
-                eventsAPIApiController.GetAllEvents(
-                    OnSuccess: eventList =>
-                    {
-                        eventsFromAPI = eventList;
-                        OnRequestedPlacesAndEventsUpdated();
-                    },
-                    OnFail: error =>
-                    {
-                        OnRequestedPlacesAndEventsUpdated();
-                        Debug.LogError($"Error receiving events from the API: {error}");
-                    });
-            }, 0, 20, cts.Token);
+        requestAllCts?.SafeCancelAndDispose();
+        requestAllCts = CancellationTokenSource.CreateLinkedTokenSource(disposeCts.Token);
+        RequestAllFromApiAsync(requestAllCts.Token).Forget();
+
+    }
+
+    private async UniTask RequestAllFromApiAsync(CancellationToken ct)
+    {
+        const int PAGE_SIZE = 12;
+
+        (IReadOnlyList<PlaceInfo> pageOne, _ ) = await placesAPIService.GetMostActivePlaces(0, PAGE_SIZE, ct);
+        (IReadOnlyList<PlaceInfo> pageTwo, _ ) = await placesAPIService.GetMostActivePlaces(1, PAGE_SIZE, ct);
+        placesFromAPI.Clear();
+        placesFromAPI.AddRange(pageOne);
+        placesFromAPI.AddRange(pageTwo);
+
+        eventsAPIApiController.GetAllEvents(
+        OnSuccess: eventList =>
+        {
+            eventsFromAPI = eventList;
+            OnRequestedPlacesAndEventsUpdated();
+        },
+        OnFail: error =>
+        {
+            OnRequestedPlacesAndEventsUpdated();
+            Debug.LogError($"Error receiving events from the API: {error}");
+        });
     }
 
     internal void OnRequestedPlacesAndEventsUpdated()
     {
         friendsTrackerController.RemoveAllHandlers();
 
-        List<PlaceCardComponentModel> trendingPlaces = PlacesAndEventsCardsFactory.ConvertPlaceResponseToModel(FilterTrendingPlaces());
+        List<PlaceCardComponentModel> trendingPlaces = PlacesAndEventsCardsFactory.ConvertPlaceResponseToModel(placesFromAPI, DEFAULT_NUMBER_OF_TRENDING_PLACES);
         List<EventCardComponentModel> trendingEvents = PlacesAndEventsCardsFactory.CreateEventsCards(FilterTrendingEvents(trendingPlaces.Count));
         view.SetTrendingPlacesAndEvents(trendingPlaces, trendingEvents);
 
         view.SetFeaturedPlaces(PlacesAndEventsCardsFactory.ConvertPlaceResponseToModel(FilterFeaturedPlaces()));
         view.SetLiveEvents(PlacesAndEventsCardsFactory.CreateEventsCards(FilterLiveEvents()));
     }
 
-    internal List<PlaceInfo> FilterTrendingPlaces() => placesFromAPI.Take(DEFAULT_NUMBER_OF_TRENDING_PLACES).ToList();
     internal List<EventFromAPIModel> FilterLiveEvents() => eventsFromAPI.Where(x => x.live).Take(DEFAULT_NUMBER_OF_LIVE_EVENTS).ToList();
     internal List<EventFromAPIModel> FilterTrendingEvents(int amount) => eventsFromAPI.Where(e => e.highlighted).Take(amount).ToList();
     internal List<PlaceInfo> FilterFeaturedPlaces()
```

Comment:
Please rethink this method: it looks like it returns a new list for nothing: as it is used just for iterating over. I think you can merge filtering with the iteration, for example, just use a cached `Predicate`

Reply:
I did my best, the underlying code is so ugly and convoluted that it was hard to simplify it. At least we are avoiding allocations now